go/cli/mcap/cmd/filter.go

```diff
@@ -207,6 +204,17 @@ func filter(
 			if errors.Is(err, io.EOF) {
 				return nil
 			}
+			if opts.recover && errors.Is(err, io.ErrUnexpectedEOF) {
+				fmt.Println("Input file was truncated.")
+				return nil
+			}
+			if opts.recover && token == mcap.TokenInvalidChunk {
+				fmt.Printf("Invalid chunk encountered, skipping: %s\n", err)
+				token, data, err = lexer.Next(buf)
+				fmt.Printf("NEXT ERR: %s\n", err)
```

Comment:
this line needs refinement. if these should be reported, maybe collect and report at the end?

Reply:
we should probably also change the printing in this patch to use stderr like line 189 does

Reply:
oops, left this debug stuff in, thanks

Reply:
My rationale for printf was that if `mcap recover` is intended to be a recovery tool, it's okay to print errors to stdout. But can change to stderr if you prefer.